model/qti/metadata/AbstractMetadataService.php

```diff
@@ -52,6 +54,20 @@ abstract class AbstractMetadataService extends ConfigurableService
      */
     protected $instances;
 
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    public function __construct($options = [], LoggerInterface $logger = null)
+    {
+        parent::__construct($options);
+
+        $this->logger = ($logger === null
+            ? common_Logger::singleton()->getLogger()
+            : $logger);
```

Comment:
We can get log from '$this->getServiceManager()->get(LoggerService::SERVICE_ID)', no? 

Why do we need to override the constructor?

Reply:
This domain service does still use `getOption()` in some places and there are at least two classes that inherit from it, so I think we need to keep extending from `ConfigurableService` to keep BC. But, although it extends from deprecated classes, we can try to avoid coupling it even more with the legacy `ConfigurableService` & `ServiceManager` classes while keeping backward compatibility with the former way of passing `$options`.

I'm using the singleton to get the logger by default, but I'm passing a mock for it in tests. Making the constructor to receive a `LoggerInterface` (instead of getting it from the `LoggerService` or the DI container) also serves to try to avoid coupling this class even more with the legacy `ConfigurableService` it extends from.

However, we may pass the logger in the `$options` array as well or get it from the service manager (and tweak a bit the unit tests to mock the SM deps) to lower the changes done in this class. Do you think that would be better?

Reply:
_(In fact, the sole purpose of the changes in this class is to have the logger passed though the constructor, so technically they can be reverted without affecting the fix)_

Reply:
I did not get, this class uses `extends ConfigurableService`, so we can call the container as I mentioned and not need for constructor override.